x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/CrossClusterAccessAuthenticationService.java

```diff
@@ -71,18 +73,17 @@ public void authenticate(final String action, final TransportRequest request, fi
             assert ApiKey.Type.CROSS_CLUSTER == apiKeyCredentials.getExpectedType();
             // authn must verify only the provided api key and not try to extract any other credential from the thread context
             authcContext = authenticationService.newContext(action, request, apiKeyCredentials);
+            var signature = crossClusterAccessHeaders.signature();
+
+            // Always validate a signature if provided
```

Comment:
I don't have a specific suggestion but this comment threw me a little because I was worried that we could accept a certificate identity (and verify it against the API key) but then not actually require that the request was signed.

That's not true because `signature` is an instance of `X509CertificateSignature` so everything is contained together, and it's not possible to have `signature == null` and somehow have a certificate identity.

_But_ if there's a way to make that clearer in this code, it would be good.

One option would be to rename the `signature` var to be `signingInfo`.